prow/gerrit/adapter/adapter.go

```diff
@@ -278,15 +278,22 @@ func (c *Controller) ProcessChange(instance string, change client.ChangeInfo) er
 	case client.New:
 		presubmits := c.config().Presubmits[cloneURI.String()]
 		presubmits = append(presubmits, c.config().Presubmits[cloneURI.Host+"/"+cloneURI.Path]...)
-		for _, presubmit := range presubmits {
-			if shouldRun, err := presubmit.ShouldRun(change.Branch, changedFiles, false, false); err != nil {
-				return fmt.Errorf("failed to determine if presubmit %q should run: %v", presubmit.Name, err)
-			} else if shouldRun {
-				jobSpecs = append(jobSpecs, jobSpec{
-					spec:   pjutil.PresubmitSpec(presubmit, refs),
-					labels: presubmit.Labels,
-				})
-			}
+		var filters []filter
+		filter, err := messageFilter(c.lastUpdate, change, presubmits)
+		if err != nil {
+			logger.WithError(err).Warn("failed to create filter on messages for presubmits")
```

Comment:
Do we append `nil` in this case to the list of filters?

Reply:
ahh, good catch! done.